Mathlib/Topology/ContinuousFunction/FunctionalCalculus.lean

```diff
@@ -0,0 +1,676 @@
+for `Topology.ContinuousFunction.FunctionalCalculus`. -/
+noncomputable irreducible_def cfc (a : A) (f : R → R) : A :=
+  if h : p a ∧ ContinuousOn f (spectrum R a)
+    then cfcSpec h.1 ⟨_, h.2.restrict⟩
+    else 0
+
+variable (a : A)
+
+lemma cfc_apply (f : R → R) (ha : p a := by cfc_tac)
+    (hf : ContinuousOn f (spectrum R a) := by cfc_cont_tac) :
+    cfc a f = cfcSpec (a := a) ha ⟨_, hf.restrict⟩ := by
+  rw [cfc_def, dif_pos ⟨ha, hf⟩]
+
+lemma cfc_apply_of_not_and {f : R → R} (ha : ¬ (p a ∧ ContinuousOn f (spectrum R a))) :
+    cfc a f = 0 := by
+  rw [cfc_def, dif_neg ha]
+
+lemma cfc_apply_of_not_predicate {f : R → R} (ha : ¬ p a) :
+    cfc a f = 0 := by
+  rw [cfc_def, dif_neg (not_and_of_not_left _ ha)]
+
+lemma cfc_apply_of_not_continuous {f : R → R} (hf : ¬ ContinuousOn f (spectrum R a)) :
+    cfc a f = 0 := by
+  rw [cfc_def, dif_neg (not_and_of_not_right _ hf)]
+
+variable (R) in
+lemma cfc_id (ha : p a := by cfc_tac) : cfc a (id : R → R) = a :=
+  cfc_apply a (id : R → R) ▸ cfcSpec_map_id (p := p) ha
+
+variable (R) in
+lemma cfc_id' (ha : p a := by cfc_tac) : cfc a (fun x : R ↦ x) = a :=
+  cfc_id R a
+
+/-- The **spectral mapping theorem** for the continuous functional calculus. -/
+lemma cfc_map_spectrum (f : R → R) (ha : p a := by cfc_tac)
+    (hf : ContinuousOn f (spectrum R a) := by cfc_cont_tac) :
+    spectrum R (cfc a f) = f '' spectrum R a := by
+  simp [cfc_apply a f, cfcSpec_map_spectrum (p := p)]
+
+lemma cfc_predicate (f : R → R) (ha : p a := by cfc_tac)
+    (hf : ContinuousOn f (spectrum R a) := by cfc_cont_tac) :
+    p (cfc a f) :=
+  cfc_apply a f ▸ cfcSpec_predicate (A := A) ha _
+
+lemma cfc_congr {f g : R → R} (hfg : (spectrum R a).EqOn f g) :
+    cfc a f = cfc a g := by
+  by_cases h : p a ∧ ContinuousOn g (spectrum R a)
+  · rw [cfc_apply (ha := h.1) (hf := h.2.congr hfg), cfc_apply (ha := h.1) (hf := h.2)]
+    congr
+    exact Set.restrict_eq_iff.mpr hfg
+  · obtain (ha | hg) := not_and_or.mp h
+    · simp [cfc_apply_of_not_predicate a ha]
+    · rw [cfc_apply_of_not_continuous a hg, cfc_apply_of_not_continuous]
+      exact fun hf ↦ hg (hf.congr hfg.symm)
+
+lemma eqOn_of_cfc_eq {f g : R → R} (h : cfc a f = cfc a g) (ha : p a := by cfc_tac)
+    (hf : ContinuousOn f (spectrum R a) := by cfc_cont_tac)
+    (hg : ContinuousOn g (spectrum R a) := by cfc_cont_tac) :
+    (spectrum R a).EqOn f g := by
+  rw [cfc_apply a f, cfc_apply a g] at h
+  have := (cfcSpec_closedEmbedding (show p a from ha) (R := R)).inj h
+  intro x hx
+  congrm($(this) ⟨x, hx⟩)
+
+lemma cfc_const (r : R) (ha : p a := by cfc_tac) :
+    cfc a (fun _ ↦ r) = algebraMap R A r := by
+  rw [cfc_apply a (fun _ : R ↦ r), ← AlgHomClass.commutes (cfcSpec ha (p := p)) r]
+  congr
+
+variable (R)
+
+lemma cfc_one (ha : p a := by cfc_tac) : cfc a (1 : R → R) = 1 :=
+  cfc_apply a (1 : R → R) ▸ map_one (cfcSpec (show p a from ha))
+
+lemma cfc_const_one (ha : p a := by cfc_tac) : cfc a (fun _ : R ↦ 1) = 1 :=
+  cfc_one R a
```

Comment:
Similarly here can one of these be simp?

Reply:
It would fire if it were, see [lean4#3475](https://github.com/leanprover/lean4/issues/3475)